gnlse/visualization.py

```diff
@@ -43,26 +262,206 @@ def plot_wavelength_vs_distance(solver, WL_range=[400, 1350], ax=None,
                       where=(np.abs(solver.AW)**2 > 0)))
     WL = 2 * np.pi * c / solver.W  # wavelength grid
     WL_asc = np.flip(WL, )  # ascending order for interpolation
-    iis = np.logical_and(WL_asc > WL_range[0],
-                         WL_asc < WL_range[1])  # indices of interest
+    iio = np.logical_and(WL_asc > WL_range[0],
+                         WL_asc < WL_range[1])  # indices in order
 
-    WL_asc = WL_asc[iis]
-    lIW = lIW[:, iis]
+    WL_asc = WL_asc[iio]
+    lIW = lIW[:, iio]
 
-    interpolator = interp2d(WL_asc, solver.Z, lIW)
-    newWL = np.linspace(np.min(WL_asc), np.max(WL_asc), lIW.shape[1])
-    toshow = interpolator(newWL, solver.Z)
+    # indices of interest if no z_slice positions were given
+    if z_slice is None:
+        iis = [0, -1]
+    # indices of interest nearest to given z_slice positions
+    else:
+        iis = [np.nonzero(
+            np.min(np.abs(solver.Z - z)) == np.abs(solver.Z - z)
+        )[0][0] for z in z_slice]
 
-    ax.imshow(toshow, origin='lower', aspect='auto', cmap="magma",
-              extent=[np.min(WL_asc), np.max(WL_asc), 0, np.max(solver.Z)],
-              vmin=-40)
+    for i in iis:
+        label_i = "z = " + str(solver.Z[i]) + "m"
+        ax.plot(WL_asc, lIW[i][:], label=label_i)
+
+    ax.set_xlim([np.min(WL_asc), np.max(WL_asc)])
+    ax.set_ylim(-40)
     ax.set_xlabel("Wavelength [nm]")
-    ax.set_ylabel("Distance [m]")
+    ax.set_ylabel("Normalized Spectral Density")
+    ax.legend()
+    return ax
+
+
+def plot_delay_for_distance_slice_logarithmic(solver, time_range=None, ax=None,
```

Comment:
To think about: maybe it would be better to have one function and as an argument scale -> lin/log?

Reply:
Issue opened: https://github.com/WUST-FOG/gnlse-python/issues/21